### PR TITLE
validation: add Rust-side certificate validation helpers

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
Breakout from #8873.

### CODE REVIEW

**Comment by tetsuo-cpp** [comment]
@woodruffw This should be good to go. Can you please give this a quick review? Then we can hand the review off to Alex and Paul.

**Comment by tetsuo-cpp** [comment]
I have some WIP to breakout `cryptography-x509-validation/src/policy/extension.rs` but some of the helpers are dependent on this so let's get this in first.

**Comment by tetsuo-cpp** [comment]
As far as I can tell, the remaining CI failure looks unrelated.

**Comment by alex** [comment]
re-started it

File: src/rust/cryptography-x509-validation/src/certificate.rs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+MzAwMFowJzELMAkGA1UEBhMCVVMxGDAWBgNVBAMMD2NyeXB0b2dyYXBoeSBDQTBZ
+MBMGByqGSM49AgEGCCqGSM49AwEHA0IABBj/z7v5Obj13cPuwECLBnUGq0/N2CxS
+JE4f4BBGZ7VfFblivTvPDG++Gve0oQ+0uctuhrNQ+WxRv8GC177F+QWjEzARMA8G
+A1UdEwEB/wQFMAMBAf8wCgYIKoZIzj0EAwIDSQAwRgIhANES742XWm64tkGnz8Dn
+pG6u2lHkZFQr3oaVvPcemvlbAiEA0WGGzmYx5C9UvfXIK7NEziT4pQtyESE0uRVK
+Xw4nMqk=
+-----END CERTIFICATE-----",
+        )
+        .unwrap()
+    }
```
Comment (woodruffw) [inline src/rust/cryptography-x509-validation/src/certificate.rs:58]: Nitpick: maybe put this as a `pub(crate)` helper under `ops::tests`, like `v1_cert_pem`? But not strong opinion.

**Review by woodruffw — APPROVED** [review]
LGTM overall, a few small thoughts.

**Review by alex — APPROVED** [review]

**Pull request merged**